contracts/DCAHub/DCAHubSwapHandler.sol

```diff
@@ -229,8 +229,7 @@ abstract contract DCAHubSwapHandler is ReentrancyGuard, DCAHubConfigHandler, IDC
     uint32 _swapFee = swapFee;
 
     {
-      // TODO: Determinate if privileged availability based on msg.sender
-      _swapInformation = getNextSwapInfo(_tokens, _pairsToSwap, true);
+      _swapInformation = getNextSwapInfo(_tokens, _pairsToSwap, hasRole(PRIVILEGED_SWAPPER_ROLE, msg.sender));
```

Comment:
Not sure how to test this without exposing how `getTotalAmountToSwap` works 😅

Reply:
Damn you Solidity, why you make everything so difficult...
@0xged , any suggestions?